catkin_home/src/nav/actions/src/navClient.py

```diff
@@ -0,0 +1,44 @@
+#! /usr/bin/env python
+
+from __future__ import print_function
+
+import rospy
+# Brings in the SimpleActionClient
+import actionlib
+
+# Brings in the messages used by the navigation action, including the
+# goal message and the result message.
+import actionlib_tutorials.msg
+import actionlib_tutorial.msg
+
+def navigationClient():
+    # Creates the SimpleActionClient, passing the type of the action
+    # (NavigationAction) to the constructor.
+    client = actionlib.SimpleActionClient('navServer', actionlib_tutorial.msg.navServAction)
+
+    # Waits until the action server has started up and started
+    # listening for goals.
+    client.wait_for_server()
+
+    # Creates a goal to send to the action server.
+    goal = actionlib_tutorial.msg.navServGoal(order = "kitchen")
+
+    # Sends the goal to the action server.
+    client.send_goal(goal)
+
+    # Waits for the server to finish performing the action.
+    client.wait_for_result()
+
+    # Prints out the result of executing the action
+    return client.get_result()  # The Navigation status result
+
+if __name__ == '__main__':
+    try:
+        # Initializes a rospy node so that the SimpleActionClient can
+        # publish and subscribe over ROS.
+        rospy.init_node('nav_client_py')
+        
+        result = navigationClient()
+        print("Result:", ', '.join([str(n) for n in result.sequence]))
+    except rospy.ROSInterruptException:
+        print("program interrupted before completion", file=sys.stderr)
```

Comment:
This is simulating the main_engine behaviour right, is it necessary to keep it in the project ?

Reply:
Yes, once the main engine is ready to be tested this test file can be deleted.